include/sparrow_ipc/arrow_interface/arrow_array/private_data.hpp

```diff
@@ -1,25 +1,33 @@
 #pragma once
-
+#include <concepts>
 #include <cstdint>
+#include <span>
+#include <variant>
 #include <vector>
 
 #include "sparrow_ipc/config/config.hpp"
 
 namespace sparrow_ipc
 {
-    class non_owning_arrow_array_private_data
+    template <typename T>
+    concept ArrowPrivateData = requires(T& t)
     {
-    public:
+        { t.buffers_ptrs() } -> std::same_as<const void**>;
+        { t.n_buffers() } -> std::convertible_to<std::size_t>;
+    };
 
-        explicit constexpr non_owning_arrow_array_private_data(std::vector<std::uint8_t*>&& buffers_pointers)
-            : m_buffers_pointers(std::move(buffers_pointers))
-        {
-        }
+    class arrow_array_private_data
+    {
+    public:
+        using optionally_owned_buffer = std::variant<std::vector<uint8_t>, std::span<const uint8_t>>;
+        explicit arrow_array_private_data(std::vector<optionally_owned_buffer>&& buffers);
```

Comment:
Do we mix owned and non-owned buffers? Otherwise a variant of vectors might be more suitable here.